rfc/rfc006-distributed-registry.md

```diff
@@ -323,18 +329,62 @@ The care organisation refers to it:
   "id": "did:nuts:abc",
   "service": [
     {
-      "id": "did:nuts:abc#NutsCompoundService-1",
+      "id": "did:nuts:abc#F1Dsgwngfdg3SH6TpDv0Ta1aOE",
       "type": "NutsCompoundService",
       "serviceEndpoint": {
-        "oauth": "did:nuts:123#NutsOAuth",
-        "fhirEndpoint": "did:nuts:123#NutsFHIR"
+        "oauth": "did:nuts:123?type=oauth",
+        "fhirEndpoint": "did:nuts:123?type=fhir"
       }
     }
   ]
 }
 ```
 
-## 5. Supported Cryptographic Algorithms and Key Types
+## 5. Conflict resolution
+
+§3.4 of [RFC004](rfc004-verifiable-transactional-graph.md) requires each payload type to describe how conflicts are resolved when parallel transactions are encountered.
+A DID Document is not a CRDT, but its contents can be. The paragraphs below describe how each of the elements should be treated in case of a conflict.
+First we describe the mechanism of detecting and resolving conflicts. DID Documents do not refer to their previous version in any way. 
+The transactions described in [RFC004](rfc004-verifiable-transactional-graph.md) do refer to previous transactions. This system can be used to detect conflicts and to resolve them.
+For DID Documents additional transactional requirements are added:
+
+* Updates to DID Documents MUST refer in their transaction to the transaction of the previous version. (In addition to the HEADs)
+* In case of a conflict, the conflict can be resolved with an additional transaction referring to the all the conflicting transactions.
+
+The algorithm for conflict detection can be referred from these requirements:
+
+- A DID Document update transaction is received.
+- The current version is resolved together with the metadata (the metadata holds the originating transaction(s)).
+- If the transaction refers to all transactions present in the metadata, the update is valid and the DID Document is replaced.
```

Comment:
updating transactions must refer to those transactions that have defined the DID Document. So a transaction could refer to a previous transaction from years ago when needed.

Reply:
The confusion might be due to the line I placed the comment on, it firstly applies to:

> Updates to DID Documents MUST refer in their transaction to the transaction of the previous version. They SHOULD also refer to the latest transactions as described by [RFC005]().

I still don't understand how this is concretely implemented, other than using `prev`s? `Prev`-ing a very old transaction is unpractical because it might (or; probably) create a very long branch, because it increases the memory/CPU usage the DAG walking algorithm: merger TX's are placed on an (in-memory) backlog until all prevs are marked as processed.

Although I do have to note that I didn't test/calculate the CPU/memory usage/performance of a guesstimate real-world case (X DIDs that are updated Y times over Z time). Maybe we should discuss it in a whiteboard-session.

Another question, do we want to mix the usage of `prev`, which is intended for consistency purposes in RFC004 with a purpose of a higher layer (RFC006)? Although when we use it to implement CRDTs in this layer, it **is** used for just that, consistency purposes.

Reply:
indeed by using `prev`. I don't see how this would be a performance problem. The very old `prev` is already processed, so no harm there (unless the processing algorithm is broken).

Referencing an extra `prev` doesn't add any requirements on the network layer.